kolibri/core/assets/src/views/__tests__/FocusTrap.spec.js

```diff
@@ -0,0 +1,82 @@
+import { render, screen, fireEvent } from '@testing-library/vue';
+import userEvent from '@testing-library/user-event';
+import FocusTrap from '../FocusTrap.vue';
+import FocusTrapWrapper from './FocusTrapWrapper.vue';
+
+const renderComponent = (props = {}) => {
+  return render(FocusTrap, {
+    props: {
+      disabled: false,
+      ...props,
+    },
+  });
+};
+
+describe('FocusTrap', () => {
+  it('should emit the "shouldFocusFirstEl" element when the tab key is pressed once', async () => {
+    const { emitted } = renderComponent();
+
+    await userEvent.tab();
+    expect(emitted()).toHaveProperty('shouldFocusFirstEl');
+    expect(emitted().shouldFocusFirstEl.length).toBe(1);
+  });
+
+  it("should trap the focus ('shouldFocusFirstEl' should be emitted twice) on pressing tab twice ", async () => {
```

Comment:
I would change this to be something like "Should trap the focus and emit "shouldFocusFirstEl" if the last focusable element is focused and we focus the next element". Because we dont get "shouldFocusFirstEl" twice because we are pressing tab twice, but because we are reaching the end of the component, and to avoid losing the focus, we should focus the first element again

Reply:
Got it! Thanks.